vllm/transformers_utils/tokenizer_group/thread_tokenizer_group.py

```diff
@@ -0,0 +1,37 @@
+import threading
+from concurrent.futures import ThreadPoolExecutor
+
+from vllm.logger import init_logger
+from vllm.transformers_utils.tokenizer_group.tokenizer_group import (
+    TokenizerGroup)
+from vllm.utils import make_async
+
+logger = init_logger(__name__)
+
+
+class ThreadPoolTokenizerGroup(TokenizerGroup):
+    """A threadpool of TokenizerGroups for async tokenization."""
+
+    def __init__(self, *args, max_workers: int, **tokenizer_config):
+        super().__init__(*args, **tokenizer_config)
+        self.local = threading.local()
+
+        def init_tokenizer():
+            logger.info(
+                f"Starting tokenizer thread {threading.current_thread().name}")
+            self.local.tokenizer = TokenizerGroup(*args, **tokenizer_config)
+
+        self.executor = ThreadPoolExecutor(
+            max_workers=max_workers,
+            thread_name_prefix='tokenizer_thread',
+            initializer=init_tokenizer,
+        )
+
+        self.encode_async = make_async(self._encode_local, self.executor)
+
+    def _encode_local(self, *args, **kwargs):
+        return self.local.tokenizer.encode(*args, **kwargs)
+
+    def encode(self, *args, **kwargs):
```

Comment:
let's do full signature instead of *args/**kwargs, makes it easier to read the code

Reply:
@Yard1 let me try to make a case for this here, happy to change if you aren't convinced :)

The actual args are kind of irrelevant here since it's essentially a pass-through within a specific implementation. Calling code is always working with the methods in the abstract superclass.

So using *args/**kwargs here to me is actually clearer because you can more easily see that it's a pass-through without the verbose boilerplate, and additionally does not need to be kept in sync if the specific args change.

Reply:
I see your point - seeing as this is a stylistic choice, it would probably be best for the authors to weigh in and set a precedent to follow @WoosukKwon @zhuohan123